### PR TITLE
feat: [FFM-2239]: Adding plan enforcement tooltips to new targeting rules page 

### PR DESCRIPTION
**Pull request by chrisblakely01**
### Stuff in Scope:
- Adds plan enforcement tooltips to the inputs in the targeting rules page 
- Adds RBAC/plan enforced menu replacing "Add New Targeting" button
- Fixes an issue where the top 
- Some UICORE components dont support tooltips. This PR adds a `DisabledFeatureTooltip` component that can be wrapped around any component to display a tooltip where necessary
- To support this a new hook `useIsFeatureDisabled` has been created, exposing some properties such as if the feature is disabled by plan enforcement, or permission etc. Previously this code was scattered in the components 

### Out of Scope
- Plan enforcement tooltips will not appear over percentage rollout inputs (handled later in separate ticket as there are open questions)
- An issue with colors in the "Add New Targeting" button. This will fall under another "colors" ticket coming later

##### Jira Links:

https://harness.atlassian.net/browse/FFM-2239

##### Screenshots:
Screenshot 2022-04-04 at 14 52 13
Screenshot 2022-04-04 at 14 52 07
Screenshot 2022-04-04 at 14 51 54
Screenshot 2022-04-04 at 14 51 49

Screenshot 2022-04-01 at 13 51 58

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by chrisblakely01** [comment]
retrigger titlecheck

**Comment by chrisblakely01** [comment]
trigger cypress

File: src/modules/75-cf/components/EditFlagTabs/TabTargeting.tsx [hunk 1]
```diff
@@ -80,7 +80,7 @@ const TabTargeting: React.FC<TabTargetingProps> = props => {
       accountIdentifier={accountIdentifier}
       environmentIdentifier={environmentIdentifier}
     >
-      <Layout.Vertical padding={{ left: 'xxlarge', right: 'large', bottom: 'large' }} spacing="medium">
+      <Layout.Vertical padding={{ left: 'xxlarge', right: 'large', bottom: 'large', top: 'xlarge' }} spacing="medium">
```
Comment (chrisblakely01) [inline src/modules/75-cf/components/EditFlagTabs/TabTargeting.tsx:83]: Fix for QA bug

File: src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/TargetingRulesTab.tsx [hunk 2]
```diff
@@ -170,7 +171,8 @@ const TargetingRulesTab = ({
     refetchFlag
   })
 
-  const isLoading = patchFeatureLoading || refetchFlagLoading || targetsLoading || segmentsLoading
+  const { featureEnabled } = useIsFeatureDisabled()
```
Comment (knagurski) [inline src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/TargetingRulesTab.tsx:174]: The naming seems a bit off here. The hook is called `Disabled`, but everything it returns is about being `Enabled`

File: src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/components/disabled-feature-tooltip/DisabledFeatureTooltip.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+  // if not a permission related, must be plan enforced related
+  return <FeatureWarningTooltip featureName={FeatureIdentifier.MAUS} />
+}
+
+export const DisabledFeatureTooltip: FC<{ fullWidth?: boolean }> = ({ children, fullWidth = false }) => {
+  const { featureEnabled } = useIsFeatureDisabled()
+
+  if (featureEnabled) {
+    return <> {children}</>
```
Comment (knagurski) [inline src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/components/disabled-feature-tooltip/DisabledFeatureTooltip.tsx:36]: Extra space
```suggestion
    return <>{children}</>
```

File: src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/TargetingRulesTab.tsx [hunk 4]
```diff
@@ -186,12 +188,16 @@ const TargetingRulesTab = ({
...
-              <Layout.Vertical spacing="small" padding={{ left: 'xlarge', right: 'xlarge' }}>
+              <Layout.Vertical
+                spacing="small"
+                padding={{ left: 'xlarge', right: 'xlarge' }}
+                className={css.flagRulesSection}
+              >
                 <Card elevation={0}>
                   <FlagToggleSwitch
-                    disabled={isLoading}
+                    disabled={disabled}
```
Comment (crystal-fong) [inline src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/TargetingRulesTab.tsx:200]: Do the components within `FormikForm` still needs to be marked `disabled` when it's already marked in line 191?

File: src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/components/disabled-feature-tooltip/DisabledFeatureTooltip.tsx (hunk 3 above)
Comment (chrisblakely01) [reply src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/components/disabled-feature-tooltip/DisabledFeatureTooltip.tsx:36]: fixed

File: src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/hooks/useIsFeatureDisabled.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    },
+    [activeEnvironment]
+  )
+
+  const isFreePlan = license.licenseInformation.CF?.edition === 'FREE'
+
+  return {
+    enabledByPermission: hasPermission,
+    enabledByPlanEnforcement,
+    isFreePlan,
```
[the remark this answers is not in the document]
Comment (chrisblakely01) [reply src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/hooks/useIsFeatureDisabled.tsx:49]: it was - then I realised the `useFeature` stuff the PLG people have made takes into account the plans so this check was redundant and I forgot to remove. I will do this

File: src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/TargetingRulesTab.tsx (hunk 2 above)
Comment (chrisblakely01) [reply src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/TargetingRulesTab.tsx:174]: I've renamed to `useFeatureEnabled` - naming is hard :/

File: src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/TargetingRulesTab.tsx (hunk 4 above)
Comment (chrisblakely01) [reply src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/TargetingRulesTab.tsx:200]: some of them do yes, and they aren't wired into the formik context like the other components are

File: src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/TargetingRulesTab.tsx (hunk 2 above)
Comment (knagurski) [reply src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/TargetingRulesTab.tsx]: #UnderstatementOfTheYear

**Review by crystal-fong — APPROVED** [review]
Nice one! Things looks so much more simpler now 🔥

**Review by kylekstewart — APPROVED** [review]

**Pull request merged**